src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Book get(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                Integer price = resultSet.getObject("price", Integer.class);
+
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a connection to the DB", e);
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                Integer price = resultSet.getObject("price", Integer.class);
+
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find the book for id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement statement = connection.prepareStatement(sql)){
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at leas one row, but inserted 0 rows.");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a connection to the DB", e);
```

Comment:
The exception message 'Can not create a connection to the DB' might be misleading here. Consider changing it to something more specific, like 'Can not update book: ' + book.